xds/src/main/java/io/grpc/xds/XdsClientMetricReporterImpl.java

```diff
@@ -120,21 +118,22 @@ public void accept(BatchRecorder recorder) {
     }, CONNECTED_GAUGE, RESOURCES_GAUGE);
   }
 
-  @Override
-  public void close() {
+  void close() {
     if (gaugeRegistration != null) {
       gaugeRegistration.close();
     }
   }
 
   void reportCallbackMetrics(BatchRecorder recorder) {
-    if (callbackMetricReporter == null) {
-      // Instantiate only if not injected
-      callbackMetricReporter = new CallbackMetricReporterImpl(recorder);
-    }
+    CallbackMetricReporter callbackMetricReporter = createCallbackMetricReporter(recorder);
     try {
-      reportResourceCounts(callbackMetricReporter);
-      reportServerConnections(callbackMetricReporter);
+      SettableFuture<Void> ret = this.xdsClient.reportResourceCounts(
+          callbackMetricReporter);
+      // Normally this shouldn't take long, but adding a timeout to avoid indefinite blocking
+      Void unused = ret.get(5, TimeUnit.SECONDS);
+      SettableFuture<Void> ret1 = this.xdsClient.reportServerConnections(callbackMetricReporter);
```

Comment:
Run the two concurrently (start both, and then get() both), so we don't pay any latencies twice?

Reply:
Makes sense, started both.